src/Illuminate/Foundation/Testing/Concerns/InteractsWithDeprecationHandling.php

```diff
@@ -42,6 +50,14 @@ protected function withoutDeprecationHandling()
                     throw new ErrorException($message, 0, $level, $file, $line);
                 }
             });
+
+            if (method_exists($this, 'beforeApplicationDestroyed')) {
+                $this->beforeApplicationDestroyed(function () {
+                    if (null !== $this->originalDeprecationHandler) {
+                        restore_error_handler();
```

Comment:
We need to reset `$this->originalDeprecationHandler = null;` otherwise the memory leak persists because PHPUnit TestSuiteIterator creates a new class instance for each test method in the class and keeps everything in memory until test suite is done. After this I think it's looking good.

Reply:
Done :+1: